greenlock-manager/index.js

```diff
@@ -0,0 +1,83 @@
+let createConnection = require("typeorm").createConnection;
+let Site = require("./src/entity/Site")
+
+let Manager = module.exports;
+let connectionOptions = {
+    "type": "mongodb",
+    "host": "localhost",
+    "port": 27017,
+    "database": "arweave-dns-gateway",
+    "synchronize": true,
+    "logging": false,
+    "entities": [__dirname + "/src/entity/Site.js"],
+    "migrations": [
+        __dirname +  "/src/migration/*.js"
+    ],
+    "subscribers": [
+        __dirname +  "/src/subscriber/*.js"
+    ],
+    "cli": {
+        "entitiesDir": __dirname +  "/src/entity",
+        "migrationsDir": __dirname + "/src/migration",
+        "subscribersDir": __dirname + "/src/subscriber"
+    }
+};
+
+Manager.create = function() {
+    let manager = {};
+
+    manager = {
+        get: async function({servername}) {
+            let connection = await createConnection(connectionOptions);
+            const site = await connection.mongoManager.findOne(Site.Site, {subject: servername});
+            await connection.close();
+
+            return site;
+        },
+
+        set: async function(opts) {
+            createConnection(connectionOptions).then(async connection => {
+                let site = new Site.Site();
+                site.subject = opts.subject;
+                site.altnames = opts.altnames;
+                site.renewAt = opts.renewAt;
+                site.deletedAt = opts.deletedAt;
+                
+                await connection.mongoManager.updateOne(Site.Site, {subject: opts.subject}, {$set: site}, { upsert: true });
```

Comment:
by using .mongoManager instead of .manager it is tied to mongodb instead of being able to use any database which is the desired feature for using typeorm

Reply:
as well as on L32 and L64

Reply:
But you need to specify explicitly which database you require I guess, for example the ormconfig.json takes explicitly the argument with name of database

Reply:
thats ok because it's one place to change and can be any database type but its possible to use .manager so that it will be easy to change the database type

Reply:
Done, made the changes. Got it!